gui-agent/vmside.c

```diff
@@ -980,14 +980,23 @@ static void process_xevent_configure(Ghandles * g, XID window,
     send_pixmap_grant_refs(g, window);
 }
 
-static void send_clipboard_data(libvchan_t *vchan, XID window, char *data, uint32_t len)
+static void send_clipboard_data(libvchan_t *vchan, XID window, char *data, uint32_t len, int protocol_version)
 {
     struct msg_hdr hdr;
     hdr.type = MSG_CLIPBOARD_DATA;
     hdr.window = window;
     if (len > MAX_CLIPBOARD_SIZE)
```

Comment:
This value should also take the protocol version into account.

Reply:
No, currently it doesn't - with your change in gui-common, the `MAX_CLIPBOARD_SIZE` will be 256k, so it won't enter this block for values above  65k but below 256k. But as just commented in the gui-common PR, better to keep `MAX_CLIPBOARD_SIZE` unchanged to avoid issues like this, and introduce new name for the new limit.

Reply:
This is now fixed. The value of `MAX_CLIPBOARD_SIZE` remains the previous value in GUI Protocol V1.7

The new values for Protocol v1.8 are new definitions within `qubes-gui-common`